test/transcoding/AtomicFSubEXTForOCL.ll

```diff
@@ -0,0 +1,64 @@
+;; Check that atomic_fetch_sub is translated to OpAtomicFAddEXT with negative
+;; value operand
+; RUN: llvm-as -opaque-pointers=0 %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_EXT_shader_atomic_float_add -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv -to-text %t.spv -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv --spirv-target-env=CL2.0 -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o - | FileCheck %s --check-prefixes=CHECK-LLVM-CL20
+
+; RUN: llvm-spirv --spirv-target-env=SPV-IR -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o - | FileCheck %s --check-prefixes=CHECK-LLVM-SPV
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; CHECK-SPIRV: Capability AtomicFloat32AddEXT
+; CHECK-SPIRV: Capability AtomicFloat64AddEXT
+; CHECK-SPIRV: Extension "SPV_EXT_shader_atomic_float_add"
+; CHECK-SPIRV: TypeFloat [[TYPE_FLOAT_32:[0-9]+]] 32
+; CHECK-SPIRV: TypeFloat [[TYPE_FLOAT_64:[0-9]+]] 64
+
+
+; Function Attrs: convergent norecurse nounwind
+define dso_local spir_func void @test_atomic_float(float addrspace(1)* %a) local_unnamed_addr #0 {
+entry:
+  ; CHECK-SPIRV: 7 AtomicFAddEXT [[TYPE_FLOAT_32]]
```

Comment:
Could you please add a check that would verify whether value is really negated in SPIRV? If `CHECK-LLVM-SPV` would start failing due to a missing negation then we can't easily point out whether the issue is in SPIRVWriter or SPIRVReader.

Reply:
The spirv textual format is not defined by SPIR-V spec, I am not very familiar with it. 
The spirv text is:
`3 TypeFloat 3 32`
`3 TypeFloat 14 64`
...
`4 Constant 3 12 3278176256`
`5 Constant 14 20 0 3228884992`
I know Constant 3 is the float operand of atomic_fetch_sub( , float, ) and Constant 14 for atomic_fetch_sub( , double, ), but I have no idea how to check its value. Is there some doc to give some introduction? Thanks very much.

Reply:
The textual format doesn't show floats as floats, but instead gives the binary representation in decimal form.

3278176256 (or `0xc3650000` in hex) is indeed `-229.0f`: https://float.exposed/0xc3650000 .  So I'd say it's fine to check for those decimal constants.